File: week1/pset1/mario.c
#include <cs50.h>
#include <stdio.h>

//A pyramid is made of two sides (sx and dx) which have a gap made of a 'double space' in between.

int main(void)
{
    int number = 0;
    while (number < 1 || number > 8) //checks that the number is between 1 and 8, both included.
    {
        number = get_int("Height: "); //if imput is different than int the function is called again.
    }
    for (int row = 0; row < number; row++) //for each row.
    {
        for (int s = 0; s < number; s++) //sx side of the pyramid.
        {
            if (s + 1 < number - row)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }

        }
        printf("  "); //prints a double space --> gap in between the two sides of the pyramid.
        for (int s = 0; s < number; s++) //dx side of the pyramid.
        {
            if (s < row + 1)
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

File: week6/pset6/sentimental-mario-more/mario.py
from cs50 import get_int


n = 0
while not (n > 0 and n < 9):
    n = get_int("Height: ")
    if n > 0 and n < 9:
        break

for i in range(1, n + 1):
    print((" " * (n - i)) + ("#" * i), end="  ")
    print("#" * i)
